scripts/Muon/GUI/Common/corrections_tab_widget/background_corrections_model.py

```diff
@@ -107,20 +129,85 @@ def populate_background_corrections_data(self) -> None:
-                    self._corrections_context.background_correction_data[run_group] = BackgroundCorrectionData(
-                        start_x, end_x)
+                    self._corrections_context.background_correction_data[run_group] = BackgroundCorrectionData(start_x,
+                                                                                                               end_x)
+                self._corrections_context.background_correction_data[run_group].uncorrected_counts_workspace = \
+                    StaticWorkspaceWrapper(workspace_name, retrieve_ws(workspace_name))
+
+    def reset_background_function_data(self) -> None:
+        """Resets the background functions back to zero when the correction mode is changed."""
+        for correction_data in self._corrections_context.background_correction_data.values():
+            correction_data.setup_functions()
+
+    def run_background_correction_for_all(self) -> None:
+        """Runs the background corrections for all stored domains."""
+        for correction_data in self._corrections_context.background_correction_data.values():
+            self._run_background_correction(correction_data)
+
+    def run_background_correction_for(self, run: str, group: str) -> None:
+        """Calculates the background for some data using a Fit."""
+        run_group = tuple([run, group])
+        if run_group in self._corrections_context.background_correction_data:
+            self._run_background_correction(self._corrections_context.background_correction_data[run_group])
+
+    def _run_background_correction(self, correction_data: BackgroundCorrectionData) -> None:
+        """Calculates the background for some data using a Fit."""
+        params = self._get_parameters_for_background_fit(correction_data)
+        function, fit_status, chi_squared = run_Fit(params, AlgorithmManager.create("Fit"))
+
+        self._handle_background_fit_output(correction_data, function, fit_status, chi_squared)
+
+    def _handle_background_fit_output(self, correction_data: BackgroundCorrectionData, function: IFunction,
+                                      fit_status: str, chi_squared: float) -> None:
+        """Handles the output of the background fit."""
+        if chi_squared > MAX_ACCEPTABLE_CHI_SQUARED:
+            correction_data.setup_functions()
+            correction_data.status = f"Correction skipped - chi squared is poor ({chi_squared:.3f})."
+        elif "Failed to converge" in fit_status:
+            correction_data.setup_functions()
+            correction_data.status = f"Correction skipped - {fit_status}"
+        else:
+            if self._corrections_context.selected_function == FLAT_BACKGROUND:
+                correction_data.flat_background = function.clone()
+            elif self._corrections_context.selected_function == FLAT_BACKGROUND_AND_EXP_DECAY:
+                correction_data.flat_background = function.getFunction(0).clone()
```

Comment:
is it intended that we will always override the default functions after each fit?